src/server.c

```diff
@@ -2127,6 +2127,54 @@ void loadingCron() {
     }
 }
 
+void blockingOperationStarts() {
+    updateCachedTime(0);
+    server.blocked_last_cron = server.mstime;
+}
+
+void blockingOperationEnds() {
+    server.blocked_last_cron = 0;
+}
+
+/* Do some of the tasks of the serverCron that are needed while we're blocked. */
+void whileBlockedCron() {
```

Comment:
Should we merge this with loadingCron? It looks like you took the code from there anyways. I guess we don't care about updating memory, but it would be nice to consolidate it in one function.

Reply:
I thought about that too and decided against it, but i'm not certain.
the loadingCron currently has an `if` rather than a `while`, so if we integrate them, come portions of this function need to run only once (if the loop run at least one cycle).

However, the other difference is that although both functions are called during loading, one is also called during busy scripts (indicated by their names), so if we merge them, we might want to add some conditions to avoid certain parts when `!server.loading`.
what do you think? should i still merge them?

@guybe7 would like to hear your opinion too.

Reply:
@oranagra i vote for merging them